### PR TITLE
Universally unique identifier on Isar

### LINKED ISSUE #301: Consider a better unique identifier of robots
**Describe the improvement you would like to see**
The current unique identifier of a robot that is used across ISAR and flotilla is currently the name, saved as `robot.RobotId` in ISAR and `robot.Name` in flotilla. Consider to differ between name and Id in ISAR and use a better id as the unique identifier. Be aware that the Id should be provided manually as a config/env variable and should not be generated automatically when starting ISAR or flotilla.

**How will this change existing functionality?**
No change - just hopefully less likely to avoid confusion of robots. 

**How will this improvement affect the current Threat Model?**
No change.

### PR DESCRIPTION
**Pull request by monkeyysups**
Closes #301

### CODE REVIEW

File: src/isar/config/settings.py [hunk 1]
```diff
@@ -183,8 +183,11 @@ class Settings(BaseSettings):
     # Type of robot ISAR is monitoring
     ROBOT_TYPE: str = Field(default="robot")
 
-    # Name or unique ID of robot
-    ROBOT_ID: str = Field(default="R2-D2")
+    # Name of robot
+    ROBOT_NAME: str = Field(default="R2-D2")
+
+    # Isar id universally unique identifier (uuni)
```
Comment (aestene) [inline src/isar/config/settings.py:189]:
```suggestion
    # Isar id universally unique identifier (uuid)
```

File: src/isar/models/mission/mission.py [hunk 2]
```diff
@@ -115,10 +115,10 @@ def set_unique_id_and_metadata(self) -> None:
 
     def _set_unique_id(self) -> None:
         plant_short_name: str = settings.PLANT_SHORT_NAME
-        robot_id: str = settings.ROBOT_ID
+        isar_id: str = settings.ISAR_ID
         now: datetime = datetime.utcnow()
         self.id = (
-            f"{plant_short_name.upper()}{robot_id.upper()}"
+            f"{plant_short_name.upper()}{isar_id.upper()}"
```
Comment (aestene) [inline src/isar/models/mission/mission.py:121]: This is one example where we should still use the robot name I think.

What this function does is create the unique identifier for the mission which is currently running. It is then used as a folder name etc. for uploads. Additional guarantees for uniqueness in this one is by using a datetime object as part of the name.

Comment (aestene) [reply src/isar/models/mission/mission.py:121]: So change `ISAR_ID` to `ROBOT_NAME` for this function.

File: src/isar/modules.py [hunk 3]
```diff
@@ -183,7 +183,7 @@ def provide_task_selector(self) -> TaskSelectorInterface:
     "queues": (QueuesModule, "required"),
     "request_handler": (RequestHandlerModule, "required"),
     "robot_package": (RobotModule, settings.ROBOT_PACKAGE),
-    "robot_id": (RobotModule, settings.ROBOT_ID),
+    "isar_id": (RobotModule, settings.ISAR_ID),
```
Comment (aestene) [inline src/isar/modules.py:186]:
```suggestion
    "isar_id": (RobotModule, settings.ISAR_ID),
    "robot_name": (RobotModule, settings.ROBOT_NAME),
```

File: src/isar/storage/utilities.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def construct_metadata_file(
             "mission_id": metadata.mission_id,
             "plant_name": metadata.plant_name,
             "mission_date": metadata.mission_date,
-            "robot_id": metadata.robot_id,
+            "isar_id": metadata.isar_id,
```
Comment (aestene) [inline src/isar/storage/utilities.py:49]:
```suggestion
            "isar_id": metadata.isar_id,
            "robot_name": metadata.robot_name,
```

**Review by aestene — CHANGES_REQUESTED** [review]
See comments

File: src/isar/models/mission/mission.py (hunk 2 above)
Comment (monkeyysups) [reply src/isar/models/mission/mission.py]: done

File: src/isar/storage/utilities.py (hunk 4 above)
Comment (monkeyysups) [reply src/isar/storage/utilities.py:49]: done

File: src/isar/config/settings.py [hunk 5]
```diff
@@ -183,8 +183,11 @@ class Settings(BaseSettings):
     # Type of robot ISAR is monitoring
     ROBOT_TYPE: str = Field(default="robot")
 
-    # Name or unique ID of robot
-    ROBOT_ID: str = Field(default="R2-D2")
+    # Name of robot
+    ROBOT_NAME: str = Field(default="R2-D2")
+
+    # Unique identifier for this ISAR instance. Note that this should be a generated UUID.
+    ISAR_ID: str = Field(default="c68b679d-308b-460f-9fe0-87eaadbd8a6e")
```
Comment (Christdej) [inline src/isar/config/settings.py:190]: Maybe change the default to something like:
"00000000-0000-0000-0000-000000000000" to make it clear it is not generated automatically

**Review by Christdej — CHANGES_REQUESTED** [review]

File: src/isar/config/settings.py (hunk 5 above)
Comment (monkeyysups) [reply src/isar/config/settings.py]: Done

File: src/isar/config/settings.env [hunk 6]
```diff
@@ -14,7 +14,8 @@ ISAR_MQTT_SSL_ENABLED = true
 
 ISAR_AUTHENTICATION_ENABLED = false
 
-ISAR_ROBOT_ID = R2-D2
+ISAR_ROBOT_NAME = R2-D2
+ISAR_ISAR_ID = c68b679d-308b-460f-9fe0-87eaadbd8a6e
```
Comment (Christdej) [inline src/isar/config/settings.env:18]: These two does not need to be here as they are defaulted in `settings.py`

File: src/isar/config/settings.py [hunk 7]
```diff
@@ -183,8 +183,11 @@ class Settings(BaseSettings):
     # Type of robot ISAR is monitoring
     ROBOT_TYPE: str = Field(default="robot")
 
-    # Name or unique ID of robot
-    ROBOT_ID: str = Field(default="R2-D2")
+    # Name of robot
+    ROBOT_NAME: str = Field(default="R2-D2")
```
Comment (Christdej) [inline src/isar/config/settings.py:187]: Can we change the default robot name to something a little more clear is the default like
"default"?

**Review by Christdej — CHANGES_REQUESTED** [review]

File: src/isar/config/settings.py (hunk 7 above)
Comment (aestene) [reply src/isar/config/settings.py:187]: `Placebot`

File: src/robot_interface/robot_interface.py [hunk 8]
```diff
@@ -118,13 +118,15 @@ def initialize(self, params: InitializeParams) -> None:
         raise NotImplementedError
 
     @abstractmethod
-    def get_telemetry_publishers(self, queue: Queue, robot_id: str) -> List[Thread]:
+    def get_telemetry_publishers(
+        self, queue: Queue, robot_name: str, isar_id: str
+    ) -> List[Thread]:
```
Comment (aestene) [inline src/robot_interface/robot_interface.py:123]:
```suggestion
    def get_telemetry_publishers(
        self, queue: Queue, isar_id: str, robot_name: str
    ) -> List[Thread]:
```

File: tests/mocks/robot_interface.py [hunk 9]
```diff
@@ -53,7 +53,9 @@ def get_inspections(self, step: InspectionStep) -> Sequence[Inspection]:
     def initialize(self, params: InitializeParams) -> None:
         return
 
-    def get_telemetry_publishers(self, queue: Queue, robot_id: str) -> List[Thread]:
+    def get_telemetry_publishers(
+        self, queue: Queue, robot_name: str, isar_id: str
+    ) -> List[Thread]:
```
Comment (aestene) [inline tests/mocks/robot_interface.py:58]:
```suggestion
    def get_telemetry_publishers(
        self, queue: Queue, isar_id: str, robot_name: str
    ) -> List[Thread]:
```

File: src/isar/config/settings.py (hunk 7 above)
Comment (monkeyysups) [reply src/isar/config/settings.py]: done

File: src/isar/config/settings.env (hunk 6 above)
Comment (monkeyysups) [reply src/isar/config/settings.env]: done

**Review by GodVenn — APPROVED** [review]
LGTM

File: src/isar/config/settings.py (hunk 1 above)
Comment (monkeyysups) [reply src/isar/config/settings.py]: done

**Review by Christdej — APPROVED** [review]
LGTM

File: src/isar/modules.py (hunk 3 above)
Comment (monkeyysups) [reply src/isar/modules.py:187]: done

File: src/robot_interface/robot_interface.py (hunk 8 above)
Comment (monkeyysups) [reply src/robot_interface/robot_interface.py]: done

**Review by aestene — APPROVED** [review]
LGTM.

I don't see a breaking change towards Flotilla as we are still using `robot_name` however we need to implement in Flotilla the switch to using `isar_id`instead.

**Pull request merged**